### PR TITLE
Drop python 3.8, add support for 3.12, 3.13

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -87,21 +87,12 @@ jobs:
...
 
-          - os: ubuntu-latest
-            python-version: 3.11-dev
-            tox-env: devel
-          - os: windows-latest
-            python-version: 3.11-dev
-            tox-env: devel
-          - os: macos-latest
-            python-version: 3.11-dev
-            tox-env: devel
```
Comment (BeyondEvil) [inline .github/workflows/tests.yml:104]: Why were these removed?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -163,12 +154,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "pypy3.9"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "pypy3.9"]
         include:
           - python-version: "3.10"
             with-coverage: true
-          - python-version: 3.11-dev
-            tox-env: devel
```
Comment (BeyondEvil) [inline .github/workflows/tests.yml:171]: Dito.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -233,10 +222,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "pypy3.9"]
-        include:
-          - python-version: 3.11-dev
-            tox-env: devel
```
Comment (BeyondEvil) [inline .github/workflows/tests.yml:239]: Dito.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (Pierre-Sassoulas) [reply .github/workflows/tests.yml:104]: It's included in the new matrice with 3.11 (or 3.14-dev when it's going to be available), right ?

Comment (BeyondEvil) [reply .github/workflows/tests.yml:104]: 3.11 yes.

But `devel` is a special target for tox where we run the suite agains pytest main-branch: https://github.com/pytest-dev/pytest-html/blob/master/tox.ini#L32

Comment (BeyondEvil) [reply .github/workflows/tests.yml:104]: So set it to `3.13`.

**Comment by BeyondEvil** [comment]
Most errors are related to `pkg_resources` which I guess is entirely removed in 3.12.

However, the code:

```python
    except AttributeError:
        # Needed for python < 3.9
        return pkg_resources.resource_string(
            "pytest_html", os.path.join("assets", "style.css")
        ).decode("utf-8")
```

We shouldn't hit.

So something is causing an attribute error here: https://github.com/pytest-dev/pytest-html/blob/master/testing/test_unit.py#L22

**Comment by Pierre-Sassoulas** [comment]
Thanks for the review @BeyondEvil ! I did not manage to install locally yet (absolute path error I don't understand). Planning to work on this in the next few days and hopefully be able to reproduce the CI fail locally if adding setuptools doesn't work.

**Comment by BeyondEvil** [comment]
> Thanks for the review @BeyondEvil ! I did not manage to install locally yet (absolute path error I don't understand). Planning to work on this in the next few days and hopefully be able to reproduce the CI fail locally if adding setuptools doesn't work.

Ugh... windows builds are always sooo slow.

Anyway, I think it's fixed.

Re-add the `devel` stuff and I think we're good to merge. 👍 

Thanks for the PR! ❤️ 🙏

**Comment by Pierre-Sassoulas** [comment]
Nice ! Will do. I plan to merge all the dependabot one next, there's a bunch of pending cve warning that stressed me out :D

**Comment by Pierre-Sassoulas** [comment]
Thank you for fixing the pipeline, I had a hard time setting up the dev env locally to troubleshoot !
